packages/flutter/lib/src/gestures/selection_recognizers.dart

```diff
@@ -0,0 +1,714 @@
+
+  void incrementConsecutiveTapCountOnDown(Offset tapGlobalPosition) {
+    if (lastTapOffset == null) {
+      // If last tap offset is null then we have not started our consecutive tap count,
+      // so the consecutiveTapTimer should be null.
+      assert(consecutiveTapTimer == null);
+      consecutiveTapCount += 1;
+      lastTapOffset = tapGlobalPosition;
+    } else if (consecutiveTapTimer != null && isWithinConsecutiveTapTolerance(tapGlobalPosition)) {
+      consecutiveTapCount += 1;
+      consecutiveTapTimerStop();
+    }
+  }
+
+  void consecutiveTapReset() {
+    consecutiveTapTimer?.cancel();
+    consecutiveTapTimer = null;
+    lastTapOffset = null;
+    consecutiveTapCount = 0;
+  }
+
+  void consecutiveTapTimerStop() {
+    consecutiveTapTimer?.cancel();
+    consecutiveTapTimer = null;
+  }
+}
+
+/// An object that includes supplementary details of a tap event, such as
+/// if the shift key was pressed when the tap occured, and what the tap count
+/// is.
+class TapStatus {
+  /// Creates a [TapStatus].
+  const TapStatus({
+    required this.consecutiveTapCount,
+    required this.isShiftPressed,
+  });
+
+  /// If this tap is in a series of taps, the `consecutiveTapCount` is
+  /// what number in the series this tap is.
+  final int consecutiveTapCount;
+
+  /// Whether the shift key was pressed when this tap happened.
+  final bool isShiftPressed;
+}
+
+/// Recognizes taps and movements.
+///
+/// Takes on the responsibilities of [TapGestureRecognizer] and [DragGestureRecognizer] in one [GestureRecognizer].
+class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _ConsecutiveTapMixin {
+  /// Initialize the object.
+  ///
+  /// [dragStartBehavior] must not be null.
+  ///
+  /// {@macro flutter.gestures.GestureRecognizer.supportedDevices}
+  TapAndDragGestureRecognizer({
+    this.deadline = kPressTimeout,
+    this.preAcceptSlopTolerance = kTouchSlop,
+    this.postAcceptSlopTolerance = kTouchSlop,
+    super.debugOwner,
+    this.dragStartBehavior = DragStartBehavior.start,
+    super.kind,
+    super.supportedDevices,
+  }) : assert(dragStartBehavior != null);
+
+  /// If non-null, the recognizer will call [onTapDown] after this
+  /// amount of time has elapsed since starting to track the primary pointer.
+  ///
+  /// [onTapDown] will not be called if the primary pointer is
+  /// accepted, rejected, or all pointers are up or canceled before [deadline].
+  final Duration? deadline;
+
+  /// {@macro flutter.gestures.monodrag.DragGestureRecognizer.dragStartBehavior}
+  DragStartBehavior dragStartBehavior;
+
+  /// The maximum distance in logical pixels the gesture is allowed to drift
+  /// from the initial touch down position before the gesture is accepted.
+  ///
+  /// Drifting past the allowed slop amount causes the gesture to be rejected.
+  ///
+  /// Can be null to indicate that the gesture can drift for any distance.
+  /// Defaults to [kTouchSlop].
+  final double? preAcceptSlopTolerance;
```

Comment:
not entirely sure why this was exposed in other gesture recongizer, but you can use doc template to avoid duplicated the doc string from